util/design/lib/OtpMemMap.py

```diff
@@ -165,19 +165,34 @@ def _validate_mmap(config):
 
     offset = 0
     num_part = 0
+    part_index = OrderedDict()
     for part in config["partitions"]:
-        num_part += 1
         _validate_part(part, offset, key_names)
 
+        if part['name'] in part_index:
+            log.error('Partition name {} is not unique'.format(part['name']))
+            exit(1)
+
         # Loop over items within a partition
+        num_items = 0
```

Comment:
Rather than tracking num_items explicitly like this, it might be easier to iterate over `enumerate(part['items'])` below.

Reply:
yeah this does not use enumerate since `num_items` is used outside of this loop later on...